crates/cli/src/main.rs

```diff
@@ -336,6 +338,39 @@ async fn main() -> anyhow::Result<()> {
     }))
     .await;
 
+    let dump_state = config.dump_state.clone();
+    let dump_interval = config
+        .state_interval
+        .map(Duration::from_secs)
+        .unwrap_or(Duration::from_secs(60)); // Default to 60 seconds
+    let preserve_historical_states = config.preserve_historical_states;
+
+    // If user provided a dump state path, spawn the periodic dumper
+    if let Some(dump_path) = dump_state {
+        let node_for_dumper = node.clone();
+        let mut state_dumper = PeriodicStateDumper::new(
+            node_for_dumper,
+            Some(dump_path),
+            dump_interval,
+            preserve_historical_states,
+        );
+
+        // Spawn a task to handle periodic dumping and final dump on shutdown
+        let dumper_handle = tokio::spawn(async move {
+            tokio::select! {
+                _ = signal::ctrl_c() => {
+                    tracing::trace!("received shutdown signal, shutting down");
+                },
+                _ = &mut state_dumper => {}
+            }
+
+            state_dumper.dump().await;
```

Comment:
@itegulov inspecting the state dump we always get:

`"l1BatchNumber": null,` 

This is problematic when you attempt to load that same dumped state:
```
thread 'main' panicked at crates/core/src/node/in_memory.rs:925:13:
encountered a block with no batch; this is not supposed to happen
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Very strange, this shouldn't happen indeed. Do you have an example of how I can reproduce this behaviour?

Reply:
Yes, to reproduce `./target/release/anvil-zksync --state-interval 1 --dump-state state.json fork --fork-url mainnet` 

Seems specific to forking?

Reply:
Ahhh, yes totally possible with forking. I didn't realize we save fork's latest block to local storage on initialization. Do you know if that's necessary to do at all? It kind of breaks a few invariants in our code.

Reply:
In any case feel free to remove that line and see if that resolves the issue for you. If not I can take another look

Reply:
hmm I will look to resolve in https://github.com/matter-labs/anvil-zksync/pull/502